web/Controllers/PropertyController.cs

```diff
@@ -179,9 +177,30 @@ public async Task<IActionResult> ListPropertyImages(int propertyId)
         {
             return StatusCode(403);
         }
-        ImageListViewModel imageList = await _azureStorage.ListFiles("property", propertyId);
+        var fileNames = await _azureStorage.ListFileNames("property", propertyId);
+        var imageFiles = GetFilesFromFileNames(fileNames, propertyId);
+
+        var imageList = new ImageListViewModel()
+        {
+            PropertyId = propertyId,
+            FileList = imageFiles
+        };
+        
         return View(imageList);
     }
+
+    public List<ImageFile> GetFilesFromFileNames(List<string> fileNames, int propertyId)
+    {
+        var imageFiles = new List<ImageFile>();
+        foreach (var fileName in fileNames)

```

Comment:
You can make this slightly nicer with `fileNames.Select(fileName => {/* Mapping code here*/})` rather than `foreach`